File: minegame-auth/src/main/java/top/nipuru/minegame/auth/processor/QueryUserHandler.java
package top.nipuru.minegame.auth.processor;

import top.nipuru.minegame.auth.domain.entity.User;
import top.nipuru.minegame.auth.service.UserService;
import top.nipuru.minegame.common.message.auth.QueryUserRequest;
import top.nipuru.minegame.common.processor.RequestDispatcher;
import org.springframework.stereotype.Component;

import javax.annotation.Resource;

@Component
public class QueryUserHandler implements RequestDispatcher.Handler<QueryUserRequest> {


    @Resource
    private UserService userService;

    @Override
    public void handle(RequestDispatcher.ResponseSender responseSender, QueryUserRequest request) {
        User user = userService.initUser(request.getUniqueId().toString(), request.getIp());
        QueryUserRequest.UserMessage userMessage = new QueryUserRequest.UserMessage(user.getPlayerId(), user.getDbId());
        responseSender.sendResponse(userMessage);
    }

    @Override
    public Class<QueryUserRequest> interest() {
        return QueryUserRequest.class;
    }
}

File: minegame-auth/src/main/kotlin/top/nipuru/minegame/auth/processor/QueryUserHandler.kt
package top.nipuru.minegame.auth.processor

import com.alipay.remoting.AsyncContext
import top.nipuru.minegame.auth.user.UserManager
import top.nipuru.minegame.common.message.auth.QueryUserRequest
import top.nipuru.minegame.common.message.auth.UserMessage
import top.nipuru.minegame.common.processor.RequestDispatcher

class QueryUserHandler : RequestDispatcher.Handler<QueryUserRequest> {
    override fun handle(asyncCtx: AsyncContext, request: QueryUserRequest) {
        val user = UserManager.initUser(request.uniqueId, request.ip)
        val userMessage: UserMessage = UserMessage(user.playerId, user.dbId)
        asyncCtx.sendResponse(userMessage)
    }

    override fun interest(): Class<QueryUserRequest> {
        return QueryUserRequest::class.java
    }
}
